src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp

```diff
@@ -403,6 +403,10 @@ struct layout {
             }
             return l.size;
         };
+
+        if (lhs.get_partial_shape().rank() != rhs.get_partial_shape().rank())
```

Comment:
It will impact static case too, but from the method behavior perspective {1,4} and {1,4,1,1} are not equal shapes, thus layouts must be not equal as well. So I'll run perf check for that patch to see how it impacts static cases. But in case of perf drops we'll need to fix those on the graph optimizer level and allow more optimizations for static cases there.